packages/contracts/test/framework/dao/dao-factory.ts

```diff
@@ -496,4 +519,196 @@ describe('DAOFactory: ', function () {
+      adminPluginRepoAddress = event.args.pluginRepo;
+
+      // create dao with admin plugin.
+      const adminPluginRepoPointer: PluginRepoPointer = [
+        adminPluginRepoAddress,
+        1,
+        1,
+      ];
+
+      let data = ethers.utils.defaultAbiCoder.encode(
+        adminMetadata.pluginSetupABI.prepareInstallation,
+        [ownerAddress]
+      );
+
+      let adminPluginInstallation = createPrepareInstallationParams(
+        adminPluginRepoPointer,
+        data
+      );
+      tx = await daoFactory.createDao(daoSettings, [adminPluginInstallation]);
+      event = await findEvent(tx, EVENTS.InstallationPrepared);
+      {
+        const adminFactory = await ethers.getContractFactory('Admin');
+        adminPlugin = adminFactory.attach(event.args.plugin);
+
+        const daoFactory = await ethers.getContractFactory('DAO');
+        dao = daoFactory.attach(event.args.dao);
+      }
+    });
+
+    it('installs,updates and uninstalls plugin through dao', async () => {
+      // Prepare Installation
+      let {
+        plugin,
+        preparedSetupData: {permissions, helpers},
+      } = await prepareInstallation(
+        psp,
+        dao.address,
+        [pluginSetupMockRepoAddress, 1, 1],
+        EMPTY_DATA
+      );
+
+      let DAO_INTERFACE = DAO__factory.createInterface();
+      let PSP_INTERFACE = PluginSetupProcessor__factory.createInterface();
+
+      // Prepare actions for apply Installation.
+      let applyInstallationActions = [
+        {
+          to: dao.address,
+          value: 0,
+          data: DAO_INTERFACE.encodeFunctionData('grant', [
+            dao.address,
+            psp.address,
+            ethers.utils.id('ROOT_PERMISSION'),
+          ]),
+        },
+        {
+          to: psp.address,
+          value: 0,
+          data: PSP_INTERFACE.encodeFunctionData('applyInstallation', [
+            dao.address,
+            createApplyInstallationParams(
+              plugin,
+              [pluginSetupMockRepoAddress, 1, 1],
+              permissions,
+              helpers
+            ),
+          ]),
+        },
+      ];
+
+      await expect(
+        adminPlugin.executeProposal('0x', applyInstallationActions, 0)
+      ).to.emit(psp, EVENTS.InstallationApplied);
+
+      // Prepare Update
+      const {
+        initData,
+        preparedSetupData: {
+          permissions: updatePermissions,
+          helpers: updateHelpers,
+        },
+      } = await prepareUpdate(
+        psp,
+        dao.address,
+        plugin,
+        [1, 1],
+        [1, 2],
+        pluginSetupMockRepoAddress,
+        helpers,
+        EMPTY_DATA
+      );
+
+      // Prepare actions for applyUpdate to succeed.
+      let applyUpdateActions = [
+        {
+          to: dao.address,
+          value: 0,
+          data: DAO_INTERFACE.encodeFunctionData('grant', [
+            plugin,
+            psp.address,
+            ethers.utils.id('UPGRADE_PLUGIN_PERMISSION'),
```

Comment:
do we also need to revoke this permission at the end of list of actions? and also check it?

Reply:
No need. We are just testing if plugin can be installed uninstalled updated. Anything else is irrelevant.